### PR TITLE
Moving the upgrade document to the developer section.

### PR DESCRIPTION
**Pull request by mbruzek**
This is not linked by navigation, our developer documents point to this file.

### CODE REVIEW

File: src/en/developer-upgrade-charm.md [hunk 1]
```diff
@@ -31,14 +31,14 @@ settings will not be affected by subsequent changes to the service's settings.
...
-Juju defines the [upgrade-charm hook](authors-charm-hooks.html#upgrade-charm)
+Juju defines the [upgrade-charm hook](reference-charm-hooks.html#upgrade-charm)
 for resolving differences between versions of the same charm. No notice is given
 of charm upgrades; a charm upgrade may run at any time the unit is started, and
 the only opportunity for resolution that exists occurs *after* the change has
 taken place.
 
 This is quite a tight restriction, but nonetheless valuable, so long as you can
-guarantee it'll run. However, it's important to understand that the upgrade-
+guarantee it will run. However, it's important to understand that the upgrade-
```
Comment (chuckbutler) [inline src/en/developer-upgrade-charm.md]: contraction :cop:

File: src/en/reference-charm-hooks.md [hunk 2]
```diff
@@ -75,11 +75,11 @@ implement the following logic:
...
 ### upgrade-charm
 
-`upgrade-charm` runs immediately after any [upgrade](./authors-charm-
-upgrades.html) operation that does _not_ itself interrupt an existing [error
-state](./authors-hook-errors.html). It should be used to reconcile local state
-written by some other version of the charm into whatever form it needs to take
-to be manipulated by the current version.
+`upgrade-charm` runs immediately after any
+[upgrade](./developer-upgrade-charm.html) operation that does _not_ itself
+interrupt an existing [error state](./authors-hook-errors.html). It should be
```
Comment (chuckbutler) [inline src/en/reference-charm-hooks.md]: Linking to an author doc thats likely to get moved/removed... 

stale link :cop:

**Comment by chuckbutler** [comment]
Minor feedback wrt using a link to an authors page, and poking fun at your removal of a contraction.

LGTM otherwise :+1:

File: src/en/reference-charm-hooks.md (hunk 2 above)
Comment (mbruzek) [inline src/en/reference-charm-hooks.md]: We will have to move that document over. It may not be deleted.

**Pull request merged**